app/src/main/java/com/ahmettekin/WeatherApp/service/CityService.java

```diff
@@ -42,44 +41,62 @@ public static CityService getInstance() {
     }
 
     public void writeDataLocalDatabase(String cityName, View view) {
```

Comment:
Bu fonksiyon iki iş yapıyor, solid'e ters :)
Yapmak istediğini anladım buraya parametre olarak bir şehir ismi gönderip servisten getirdiğin listenin içerisinde aratıyorsun ve öyle bir şehir varsa ekliyorsun güzel fakat bence sistem şöyle işlemeli:
 AddCityFragment create olduğunda tüm şehirlerin listesi çekilmeli ve progress bar bu liste çekilene kadar ekranda kalmalı.Şehirler çekilebildiğinde yani success olduğunda progress bar gizlenmeli ve sonrasında kullanıcı edittext'e yazıp ekle demeli.Bu şekilde olduğunda `writeDataLocalDatabase` fonksiyonu da sadece kendi işi olan veritabanına yazma işini yapabilir 👍

Reply:
Harun merhaba. Bu dediğin olay için uğraştım ama bu şekilde yapabilmem için fonksiyonun illa birşey döndürmesi gerekiyor. 
Birşey döndürünce de arka planda çalışma problemi yine ortaya çıkıyor.

Reply:
Nasıl yapacağımı bilemedim :)

Reply:
Yani şehir ismini girip id sini servisten çektiğimde aynı anda o id yi database' e eklemem gerekiyor ki sorun oluşmasın.

Reply:
ok buna müsait zamanda birlikte bakalım şimdilik pr açık kalsın 🙂 👍